src/test/java/com/pubnub/contract/subscribe/state/SubscribeState.kt

```diff
@@ -0,0 +1,8 @@
+package com.pubnub.contract.subscribe.state
+
+import com.pubnub.api.models.consumer.pubsub.MessageResult
+
+
+class SubscribeState {
+    var messages: List<MessageResult> = mutableListOf()
```

Comment:
I don't think it should be `var` or it shouldn't be `mutableListOf`, cause it doesn't make sense to have both.

In case of `val message: MutableList` maybe it's not a bad idea to eliminate any thread safety problems we can somehow have. To do that you could maybe use CopyOnWriteArrayList. I know that it's very improbable that we'll have any concurrency issues, but in tests we can pay the performance price without any issue and eliminate them completely. Maybe I'm overreacting 😵‍💫

Reply:
Interesting proposal. I will use CopyOnWriteArrayList.